test/test_endpoints.py

```diff
@@ -350,3 +381,147 @@ def test_get_transaction_trace_of_rejected():
     resp_body = resp.json()
     assert resp_body["code"] == str(StarknetErrorCode.NO_TRACE)
     assert resp.status_code == 500
+
+
+@devnet_in_background()
+def test_send_message_to_l2_deploy_execute():
+    """Test POST l1 to l2 deploy contract and execute transaction"""
+    # Deploy L1L2 contract
+    deploy_info = deploy(contract=L1L2_CONTRACT_PATH)
+
+    # Create l1 to l2 mock transaction
+    response = send_message_to_l2(
+        {
+            "l2_contract_address": deploy_info["address"],
+            "entry_point_selector": "0xC73F681176FC7B3F9693986FD7B14581E8D540519E27400E88B8713932BE01",
+            "l1_contract_address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
+            "payload": ["0x1", "0x1"],
+            "nonce": "0x0",
+        }
+    )
+
+    # Check balace of user
+    value = call(
+        function="get_balance",
+        address=deploy_info["address"],
+        abi_path=L1L2_ABI_PATH,
+        inputs=[str(USER_ID)],
+    )
+
+    assert int(value) == 1
```

Comment:
Do we need to assert some nonce here as well?

Reply:
I don't think so, this is handled by `StarknetContract` from l1 and this software component is missing in our case, we are just handling that manually.